esque/cli/commands.py

```diff
@@ -274,12 +227,85 @@ def get_consumergroups(state):
 @get.command("topics")
 @click.argument("topic", required=False, type=click.STRING, autocompletion=list_topics)
 @pass_state
-def get_topics(state, topic, o):
+def get_topics(state, topic):
     topics = TopicController(state.cluster).list_topics(search_string=topic)
     for topic in topics:
         click.echo(topic.name)
 
 
+@esque.command("transfer", help="Transfer messages of a topic from one environment to another.")
+@click.argument("topic", required=True)
+@click.option("-f", "--from", "from_context", help="Source Context", type=click.STRING, required=True)
+@click.option("-t", "--to", "to_context", help="Destination context", type=click.STRING, required=True)
+@click.option("-n", "--numbers", help="Number of messages", type=click.INT, required=True)
+@click.option("--last/--first", help="Start consuming from the earliest or latest offset in the topic.", default=False)
+@click.option("-a", "--avro", help="Set this flag if the topic contains avro data", default=False, is_flag=True)
+@click.option(
+    "-k",
+    "--keep",
+    "keep_file",
+    help="Set this flag if the file with consumed messages should be kept.",
+    default=False,
+    is_flag=True,
+)
+@pass_state
+def transfer(
+    state: State, topic: str, from_context: str, to_context: str, numbers: int, last: bool, avro: bool, keep_file: bool
+):
+    current_timestamp_milliseconds = int(round(time.time() * 1000))
+    unique_name = topic + "_" + str(current_timestamp_milliseconds)
+    group_id = "group_for_" + unique_name
+    directory_name = "message_" + unique_name
+    base_dir = Path(directory_name)
+    state.config.context_switch(from_context)
+
+    with HandleFileOnFinished(base_dir, keep_file) as working_dir:
+        number_consumed_messages = _consume_to_file(working_dir, topic, group_id, from_context, numbers, avro, last)
+
+        if number_consumed_messages == 0:
+            click.echo(click.style("Execution stopped, because no messages consumed.", fg="red"))
```

Comment:
I would state the reason for stopping the execution. No consumption is the effect. This case is probably getting hit when the topic is empty or the starting offset was chosen to high, right?

Reply:
How would you do this? Maybe checking the high_watermarks for all topics as indicator for an empty topic? What with log retention it could also be empty with a high_watermark greater 0.

Reply:
I didn't mean in that much detail. 
The user can still find out what exactly the problem was by him/herself. You could just name the possible reasons in the error message.

Reply:
ah sure